airflow/providers/amazon/aws/hooks/redshift.py

```diff
@@ -126,3 +135,101 @@ def create_cluster_snapshot(self, snapshot_identifier: str, cluster_identifier:
             ClusterIdentifier=cluster_identifier,
         )
         return response['Snapshot'] if response['Snapshot'] else None
+
+
+class RedshiftSQLHook(DbApiHook):
+    """
+    Execute statements against Amazon Redshift, using redshift_connector
+
+    This hook requires the redshift_conn_id connection. This connection must
+    be initialized with the host, port, login, password. Additional connection
+    options can be passed to extra as a JSON string.
+
+    :param redshift_conn_id: reference to
+        :ref:`Amazon Redshift connection id<howto/connection:redshift>`
+    :type redshift_conn_id: str
+
+    .. note::
+        get_sqlalchemy_engine() and get_uri() depend on sqlalchemy-amazon-redshift
+    """
+
+    conn_name_attr = 'redshift_conn_id'
+    default_conn_name = 'redshift_default'
+    conn_type = 'redshift+redshift_connector'
+    hook_name = 'Amazon Redshift'
+    supports_autocommit = True
+
+    @staticmethod
+    def get_ui_field_behavior() -> Dict:
+        """Returns custom field behavior"""
+        return {
+            "hidden_fields": [],
+            "relabeling": {'login': 'User', 'schema': 'Database'},
+        }
+
+    @cached_property
+    def conn(self):
+        return self.get_connection(self.redshift_conn_id)  # type: ignore[attr-defined]
+
+    def _get_conn_params(self) -> Dict[str, Union[str, int]]:
+        """Helper method to retrieve connection args"""
+        conn = self.conn
+
+        conn_params: Dict[str, Union[str, int]] = {}
+
+        if conn.login:
+            conn_params['user'] = conn.login
+        if conn.password:
+            conn_params['password'] = conn.password
+        if conn.host:
+            conn_params['host'] = conn.host
+        if conn.port:
+            conn_params['port'] = conn.port
+        if conn.schema:
+            conn_params['database'] = conn.schema
+
+        return conn_params
+
+    def get_uri(self) -> str:
+        """
+        Override DbApiHook get_uri method for get_sqlalchemy_engine()
+
+        .. note::
+            Value passed to connection extra parameter will be excluded
+            from returned uri but passed to get_sqlalchemy_engine()
+            by default
+        """
+        from sqlalchemy.engine.url import URL
+
+        conn_params = self._get_conn_params()
+
+        conn = self.conn
+
+        conn_type = conn.conn_type or RedshiftSQLHook.conn_type
+
+        if 'user' in conn_params:
+            conn_params['username'] = conn_params.pop('user')
```

Comment:
Can the statement in `_get_conn_params()` for `user` assignment be changed to have the `conn_params` key named "username" instead of deleting the "user" and adding the "username" key here?

From this:
```python
if conn.login:
    conn_params['user'] = conn.login
```
to:
```python
if conn.login:
    conn_params['username'] = conn.login
```

Reply:
The reason for this approach is that the method that builds the URL expects `username` while the DB-API driver expects `user`.

Reply:
I reworked this in 1b22d59c1d3f25370b81c5f414a6e129a320e0ab. The overridden `get_uri()` method is removed in exchange for `_get_sqlalchemy_uri()` which contains this `user -> username` change. `_get_sqlalchemy_uri()` is used in the overriden `get_sqlalchemy_engine()` method.

The `user -> username` key change is required for the sqlalchemy URL method, which expects `username`, while the driver itself expects `user`.

Reply:
hmm... not sure why you'd want to leave `get_uri` in place and add `_get_sqlalchemy_uri`... AFAIK, the only realon there is a `get_uri` method in hooks is for sqlalchemy.  so i'd just move your logic to get_uri...  but perhaps you had a reason i'm not seeing?

Reply:
> AFAIK, the only realon there is a get_uri method in hooks is for sqlalchemy

In this case, the approach I took in 1b22d59 with `_get_sqlalchemy_uri` doesn't make sense.
I thought I had seen `get_uri` mentioned for some other use in the docs somewhere but after taking another look I see that I was mistaken.

Reply:
I've brought `get_uri` back in 4589472, and removed `_get_sqlalchemy_uri` as it's unnecessarily confusing to have both :)